caddyconfig/httpcaddyfile/options.go

```diff
@@ -23,7 +23,29 @@ import (
 	"github.com/caddyserver/caddy/v2/modules/caddytls"
 )
 
-func parseOptHTTPPort(d *caddyfile.Dispenser) (int, error) {
+func init() {
```

Comment:
Instead of using `init` can you make this a regular function, and call it explicitly at the beginning of `evaluateGlobalOptionsBlock`?

Reply:
Hmm, I'm not sure I see the advantage to that. I'm matching the way it's done in `caddyconfig/httpcaddyfile/builtins.go`

Reply:
After a separate discussion on slack, we can dismiss this; there was a misunderstanding in the goal of this PR 😄

Reply:
Well I still think using `init()` where not completely necessary is not a great idea. In this case especially, there are already other `init()` functions, which means it all of a sudden becomes hard to know which code will get executed first.

Have a look at https://peter.bourgon.org/blog/2017/06/09/theory-of-modern-go.html for some discussion around this (and other things)

Reply:
Order of execution doesn't matter here, all the work being done is just adding closures to a map. I'm just gonna follow Matt's lead here and do it the way it was already done for regular Caddyfile directives

Reply:
FWIW I don't disagree with your take, I just think being consistent is more important here.

Reply:
Fair enough. Personally I think all the `init()`s used in the codebase should be revisited 😉

Reply:
@hairyhenderson Inits are really useful for registration functions.

Reply:
@mholt agreed, except in this case the registration is something that can be done on-demand (and the current code _is_ on-demand)...

It's not a big deal, just makes things a little harder to reason about, and can make tests a bit more awkward...